qbamfix/src/org/qcmg/qbamfix/FixBAM.java

```diff
@@ -187,48 +187,39 @@ HashMap<String, Integer> checkMate(List<SAMRecord> reads, File inBAM)throws IOEx
 	 */
 	List<SAMRecord> firstFilterRun(File output) throws IOException{
 		
-		SamReader reader = SAMFileReaderFactory.createSAMFileReader(input, null,  validation);  
-
-		SAMOrBAMWriterFactory factory;		
-		if(reader.getFileHeader().getSortOrder().equals(header.getSortOrder())) {
-			factory = new SAMOrBAMWriterFactory(header, true, output, tmpDir, 2000000, true );
-		} else {
-        		factory = new SAMOrBAMWriterFactory(header, false, output, tmpDir, 2000000, true);
-		}
-						 
-		SAMFileWriter writer = factory.getWriter();
-		
-        List<SAMRecord> badReads = new ArrayList<>();
-        
+        List<SAMRecord> badReads = new ArrayList<>();       
         String id = header.getReadGroups().get(0).getId();  
         long NumofInput = 0;
         long NumofOutput = 0;
-        boolean ok2add;
-	    	for( SAMRecord record : reader){
-	    		record.setAttribute("RG", id );
-  
-	    		if(seqLength > 0)
-	    			ok2add = (record.getReadLength() == seqLength); 
-	    		else if(seqLength == 0)
-	    			ok2add = (record.getReadLength() == record.getBaseQualityString().length());
-	    		else
-	    			ok2add = true;
-	    		
-	    		if (ok2add){
-	    			writer.addAlignment(record);	
-	    			NumofOutput ++;
-	    		} else {
-	    			badReads.add(record);
-	    		}
-	    		NumofInput ++;	
+        boolean ok2add;		
+		
+		try(SamReader reader = SAMFileReaderFactory.createSAMFileReader(input, null,  validation);) {
+			SAMWriterFactory factory = 	reader.getFileHeader().getSortOrder().equals(header.getSortOrder())?			
```

Comment:
could be simplified to:
`SAMWriterFactory factory = new SAMWriterFactory(header, reader.getFileHeader().getSortOrder().equals(header.getSortOrder()), output, tmpDir, 2000000, true )`

Reply:
yes, simplified.